backend/src/main/scala/com/softwaremill/bootzooka/user/UserService.scala

```diff
@@ -100,23 +100,23 @@ class UserService(
   def changePassword(userId: Id @@ User, currentPassword: String, newPassword: String): ConnectionIO[Unit] =
     for {
       user <- userOrNotFound(userModel.findById(userId))
-      _ <- verifyPassword(user, currentPassword)
+      _ <- verifyPassword(user, currentPassword, validationErrorMsg = "Incorrect current password")
       _ = logger.debug(s"Changing password for user: $userId")
       _ <- userModel.updatePassword(userId, User.hashPassword(newPassword))
     } yield ()
 
   private def userOrNotFound(op: ConnectionIO[Option[User]]): ConnectionIO[User] = {
     op.flatMap {
       case Some(user) => user.pure[ConnectionIO]
-      case None       => Fail.NotFound("user").raiseError[ConnectionIO, User]
+      case None       => Fail.Unauthorized("Incorrect login/email or password").raiseError[ConnectionIO, User]
```

Comment:
We should probably extract this message to a constant so that it's not repeated, and the fact that the constant is used twice will also be informative for the developer